content/concepts/peer-routing.md

```diff
@@ -3,6 +3,80 @@ title: Peer Routing
 weight: 5
 ---
 
-This article is coming soon!
+Libp2p primarily uses Kademlia DHT for peer discovery. Let's dive a little bit into how Kademlia works.
 
-Please [refer to this issue](https://github.com/libp2p/docs/issues/13) to track the progress and make suggestions.
+## Static DHT
+
+To grow intuition about the Kademlia let's assume for now that our DHT is static. No one joins DHT and no one leaves.
+
+### System Descritpion
+
+Each peer has their own peer ID. Kademlia provides an efficient lookup algorithm that locates successively “closer” nodes to any desired ID.
+
+In Kademlia peers are key-value pairs in the DHT. On practice peer ID stored as a key and [address](/concepts/addressing/) (or IP address) of a peer stored as a value.
+
+Kademlia treats peers as leaves in a binary tree. Peer IDs in leaves represented as binary numbers. Notice, that we do not need to store the whole peer id to specify the peer. We can storeƒ only their prefix if prefix is unique.
+
+Leaves on the tree that are not peers represent [content](/concepts/content-routing/) which is stored by some close nodes.
+
+![Kademlia-system](Kademlia-system.png)
+
+Red node here has prefix `001`, go from the top to the bottom by edges. Let's say it' Bob. Alice here has prefix `11101` - she is blue node. Yellow leaves are other peers. For any given node, Kademlia divides the whole binary tree into a series of successively lower subtrees that don’t contain the given node. The highest subtree consists of the half of the binary tree not containing the node. The next subtree consists of the half of the remaining tree not containing the node, and so forth. With example it'll be more clear. You can see here for node `001` subtrees are circled and consist of all nodes with prefixes `1`, `01`, `000`.
+
+The Kademlia protocol ensures that every node knows of at least one node in each of its subtrees, if that subtree contains a node. This guarantees that any node can be located by it's peer ID.
+
+Let's see how Bob can locate Alice in this example. Bob knows Alices peerID, but he needs to get the value - Alices IP address. From subtree with prefix `1` Bob knows peer with prefix `100`. He sends a message to their address, where he asks for Alices address.
+
+![1-hop](1-hop.png)
```

Comment:
Mhh, these don't seem to render in GitHubs markdown view. Do they do so on your end?

Reply:
No, they don't for me
I was doing the same thing as in [pubsub](https://github.com/libp2p/docs/blob/master/content/concepts/publish-subscribe.md) article, so it shows up [here](https://bafybeidklhiwzma7r3vufxkvoym3pq6pv2l2hnmnu5uynvpqfjnbqmd2bi.on.fleek.co/concepts/peer-routing/)